src/frequenz/client/reporting/_client.py

```diff
@@ -66,41 +66,50 @@ def is_empty(self) -> bool:
             return True
         return False
 
-    def __iter__(self) -> Iterator[MetricSample]:
+    def __iter__(self) -> Iterator[dict[Any, Any]]:
```

Comment:
Sorry, I think there was a misunderstanding. The idea was to have the protobuf message represented by a dedicated dataclass. A dict is flexible but wouldn't help us for type safety, which is important here.